probe-rs/src/vendor/st/mod.rs

```diff
@@ -45,6 +45,8 @@ impl Vendor for St {
             DebugSequence::Arm(Stm32h7::create(Stm32h7Line::H7S))
         } else if chip.name.starts_with("STM32H7") {
             DebugSequence::Arm(Stm32h7::create(Stm32h7Line::H7))
+        } else if chip.name.starts_with("STM32H747") {
```

Comment:
Becasue of `starts_with` you need to check the longer match first.

Reply:
Thanks, should be fixed now